docs/src/modules/redux/optionsReducer.js

```diff
@@ -15,7 +14,7 @@ function getPath(obj, path) {
 
 const warnOnce = {};
 
-const getT = memoize((userLanguage) => (key, options = {}) => {
```

Comment:
This only memoized creation of a function not the actual `options.t` utility. We could also inline the function in the reducer and rely on closures. Then it's more obvious that this can't be a problematic pattern unless you also consider function components declaring functions in their render method a problem.

Reply:
I think that's the objective: to memoize `options.t`. The function is created at each update in the reducer. If it's not memorized, won't it rerender all the components that have?

```jsx
const t = useSelector((state) => state.options.t);
```

Reply:
Good oppurtinity to add `useTranslate` and `useUserLanguage` then. Been meaning to abstract that selector for a while now but there was no real point. Makes more sense now.

Reply:
`useUserLanguage` is especially interesting in next 10 where we might not use redux for language handling.